src/js/color.ts

```diff
@@ -0,0 +1,3845 @@
+    if (xyz === null) {
+      return xyz!
+    }
+    ;[, x, y, z, alpha] = xyz
+  } else if (value.startsWith(FN_COLOR)) {
+    ;[, x, y, z, alpha] = parseColorFunc(value) as [
+      number,
+      number,
+      number,
+      number,
+      number,
+    ]
+  } else {
+    ;[, x, y, z, alpha] = parseColorValue(value) as [
+      number,
+      number,
+      number,
+      number,
+      number,
+    ]
+  }
+  ;[l, c, h] = convertXyzToOklch([x as number, y as number, z as number], true)
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+  return [l!, c!, h!, alpha as number]
+}
+
+/**
+ * resolve color-mix()
+ * @param {string} value - color value
+ * @param {object} [opt] - options
+ * @param {string} [opt.format] format - output format
+ * @returns {Array.<string|number>|?string}
+ *   - [cs, v1, v2, v3, alpha], '(empty)', null
+ */
+export const resolveColorMix = (
+  value: string,
+  opt: {
+    format?: string
+  } = {},
+): Array<string | number> | (string | null) => {
+  if (isString(value)) {
+    value = value.toLowerCase().trim()
+  } else {
+    throw new TypeError(`${value} is not a string.`)
+  }
+  const { format } = opt
+  const nestedItems = [] as Array<string>
+  if (!REG_MIX.test(value)) {
+    if (value.startsWith(FN_MIX) && REG_MIX_NEST.test(value)) {
+      const regColorSpace = new RegExp(`^(?:${CS_RGB}|${CS_XYZ})$`)
+      const items = value.match(REG_MIX_NEST)!
+      for (const item of items) {
+        let val = resolveColorMix(item, {
+          format: format === VAL_SPEC ? format : VAL_COMP,
+        }) as string
+        // computed value
+        if (Array.isArray(val)) {
+          const [v1, v2, v3, v4, v5] = val
+          if (v2 === 0 && v3 === 0 && v4 === 0 && v5 === 0) {
+            value = ''
+            break
+          }
+          if (regColorSpace.test(v1)) {
+            if (v5 === 1) {
+              val = `color(${v1} ${v2} ${v3} ${v4})`
+            } else {
+              val = `color(${v1} ${v2} ${v3} ${v4} / ${v5})`
+            }
+          } else if (v5 === 1) {
+            val = `${v1}(${v2} ${v3} ${v4})`
+          } else {
+            val = `${v1}(${v2} ${v3} ${v4} / ${v5})`
+          }
+        } else if (!REG_MIX.test(val)) {
+          value = ''
+          break
+        }
+        nestedItems.push(val)
+        value = value.replace(item, val)
+      }
+      if (!value) {
+        if (format === VAL_SPEC) {
+          return ''
+        }
+        return ['rgb', 0, 0, 0, 0]
+      }
+    } else if (format === VAL_SPEC) {
+      return ''
+    } else {
+      return ['rgb', 0, 0, 0, 0]
+    }
+  }
+  let colorSpace, hueArc, colorA, pctA, colorB, pctB
+  if (nestedItems.length && format === VAL_SPEC) {
+    const regColorSpace = new RegExp(`^color-mix\\(\\s*in\\s+(${CS_MIX})\\s*,`)
+    const [, cs] = value.match(regColorSpace) as [string, string]
+    if (REG_CS_HUE.test(cs)) {
+      ;[, colorSpace, hueArc] = REG_CS_HUE.exec(cs) as unknown as [
```

Comment:
why `unknown`?

Reply:
`.exec` returns a non-array type (`RegExpExecArray`) that needs to be cast to `unknown` before cast to an array